src/envoy/http/authn/authenticator_base.cc

```diff
@@ -68,7 +73,21 @@ bool AuthenticatorBase::validateX509(const iaapi::MutualTls& mtls,
 bool AuthenticatorBase::validateJwt(const iaapi::Jwt& jwt, Payload* payload) {
   std::string jwt_payload;
   if (filter_context()->getJwtPayload(jwt.issuer(), &jwt_payload)) {
-    return AuthnUtils::ProcessJwtPayload(jwt_payload, payload->mutable_jwt());
+    std::string payload_to_process = jwt_payload;
+    std::string original_payload;
+    bool found = false;
+    for (auto h : jwt.jwt_headers()) {
+      if (kExchangedTokenHeaderName == h) {
+        found = true;
+        break;
+      }
+    }
+    if (found &&
+        AuthnUtils::ExtractOriginalPayload(jwt_payload, &original_payload)) {
```

Comment:
Instead of doing string copy, you can directly put "payload_to_process" in the function. 

AuthnUtils::ExtractOriginalPayload(jwt_payload, &payload_to_process)

Reply:
If put "payload_to_process" in ExtractOriginalPayload(), ExtractOriginalPayload() may change "payload_to_process" even when ExtractOriginalPayload() returns false. Therefore,  "payload_to_process" should not be put in ExtractOriginalPayload().